### ISSUE TITLE
widget should work more like select when drop-down is closed and user types a letter

### ISSUE THREAD

**Issue by dbeusee** [select2/select2#4586]
## Steps to reproduce the issue
1. Tab into field.
2. Type a letter where a value begins with that letter
## Expected behavior and actual behavior

When I follow those steps, I see...
Nothing happens.

I was expecting...
Value should change to the next choice matching first letter.  I'm happy to do full search when the drop down is open, but when closed, it should be able to move to next value matching the letter.  Also down-arrow and up-arrow keys should move between values even when drop down is closed.
## Environment

Browsers
- [X] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System
- [X] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries
- jQuery version: 2.0.2
- Select2 version: 4.0.3
## Isolating the problem
- [ ] This bug happens [on the examples page](https://select2.github.io/examples.html)
- [ ] The bug happens consistently across all tested browsers
- [ ] This bug happens when using Select2 without other pluigns
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by gertjanketelaars** [comment]
I think this is a nice feature! 

Not sure if this should be default behavior or an option, but I have made some changes to support it. See https://github.com/gertjanketelaars/select2/commits/patch4586_open_on_key.

Is this the right approach? If so I can make a pull-request.

This issue seems a duplicate of #4514.

**Comment by alexrockt** [comment]
+1

**Comment by hughlaw** [comment]
Looks like this (sort of) functionality was lost between V3 and V4

I've put together 2 fiddles to show the differences when tabbing into a select and then typing.
V3 behaves far more like a native `select` than V4 does

[V3.5 Stable](https://jsfiddle.net/hughlaw/qkv4j92x/)

[Select2 latest version](https://jsfiddle.net/hughlaw/eyuvva7e/)

**Comment by csizmazia** [comment]
actually, this still worked in v4.0.2

**Comment by rordi** [comment]
Does not work for me with v4.0.3, tested on Chrome, Firefox, Safari. One needs to hit enter or spacebar to open the select2 dropdown and start typing/searching.

**Comment by DavidAllison** [comment]
Seeing the same thing - any chance of an update given that this looked like a regression in 4.0.3?

**Comment by alexweissman** [comment]
I'm not sure, but I think #5130 might address this issue.
